File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        
        for i, num in enumerate(nums):
            diff = target - num
            if diff in dic:
                return [i, dic[diff]]
            dic[num] = i
                

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const seen = {} // remainder: index
    
    for (const i in nums) {
        const num = nums[i]
        const remainder = target - num
        
        if (remainder in seen) return [seen[remainder], i]
        seen[num] = i
    }
};